src/Files/ServicesImplementation/Win32StorageEnumerator.cs

```diff
@@ -1,13 +1,59 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.IO;
 using Files.Backend.Services;
 using Files.Backend.ViewModels.ItemListing;
+using Files.Filesystem;
+using Microsoft.Toolkit.Uwp;
+using static Files.Helpers.NativeFindStorageItemHelper;
 
 namespace Files.ServicesImplementation
 {
     internal sealed class Win32StorageEnumerator : IStorageEnumeratorService
     {
-        public bool IsAvailable { get; }
+        private IntPtr hFile;
+        private WIN32_FIND_DATA findData;
+
+        public string WorkingDirectory
```

Comment:
I would put it in BaseListedLayoutViewModel

Reply:
This would complement what's in that view model. We just needed some way to have the path be tied to the service instance in order to enable checking if that service is supported for the file system location.

Reply:
Alternative would be to pass a path to Enumerate(string folderPath) / IsSupported(string folderPath)

Reply:
Yes! That'd make it much more flexible

Reply:
@gave92 @d2dyno1 I'm not a fan of that pattern. It seems more readable if we don't have duplicate opportunities to pass parameters for what will be the same object. 

Moreover, the way I committed means we only have to validate the path once rather than for each method that takes it for an argument.

Reply:
There's no real reason why we should keep the service in know of the current directory. The sole purpose of services is to process the provided data and return appropriate results. It shouldn't really know of the state of the navigated directory. There are some services which may know particular states, but this doesn't apply here. My intent was to create an abstract way directories are traversed, and so these enumerating services should only be supplied with desired path and should only supply back the result. There's no need to incorporate the current directory path nor any contextual data. That would be done in `BaseListedLayout` (or `ListingModel`) and shall be enough.

@duke7553

Reply:
On the other hand, I'm still thinking whether to make it transient or singleton. **If** we used the transient we'd obviously be able to use the contextual data, but that's not my intent. The question is, how will the service behave with multithreading - will singleton affect it?